src/Common/src/CoreLib/System/IO/Win32Marshal.cs

```diff
@@ -25,6 +26,8 @@ internal static Exception GetExceptionForLastWin32Error(string? path = "")
         /// </summary>
         internal static Exception GetExceptionForWin32Error(int errorCode, string? path = "")
         {
+            Debug.Assert(errorCode != Interop.Errors.ERROR_SUCCESS);
```

Comment:
This warrants a comment, as I don't think it is obvious what we're asserting and what the remedy should be. Can you give a shot at explaining why we're checking this and what you should do if you get it?

Reply:
Sure, gladly. How about this?:
```suggestion
            // ERROR_SUCCESS gets thrown when another unexpected interop call was made before checking GetLastWin32Error(). Errors have to get retrieved as soon as possible after P/Invoking to avoid this.
            Debug.Assert(errorCode != Interop.Errors.ERROR_SUCCESS);
```